### PR TITLE
HDDS-7450. Improve bucket creation message

### PR DESCRIPTION
**Pull request by chrisGe123**
https://issues.apache.org/jira/browse/HDDS-7450

### CODE REVIEW

**Comment by chrisGe123** [comment]
correct the code style issue

File: hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/rpc/RpcClient.java [hunk 1]
```diff
@@ -668,13 +669,20 @@ public void createBucket(
       builder.setDefaultReplicationConfig(defaultReplicationConfig);
     }
 
+    ReplicationType replicationType = defaultReplicationConfig == null ?
+        ReplicationType.RATIS : defaultReplicationConfig.getType();
```
Comment (adoroszlai) [inline hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/rpc/RpcClient.java:673]: I don't think the client should make assumptions about default replication on the server.  If the client does not set the bucket's default replication config, it should omit `Replication Type set to ...` from the message.

**Review by adoroszlai — CHANGES_REQUESTED** [review]
Thanks @chrisGe123 for the patch.

You can verify checkstyle by running `hadoop-ozone/dev-support/checks/checkstyle.sh` locally before creating the PR.  Please also enable the [`build-branch` workflow](https://github.com/chrisGe123/ozone/actions/workflows/post-commit.yml) in your fork to get full CI run (preferably before opening the PR).

Furthermore, please follow Ozone's [pull request template](https://github.com/apache/ozone/blob/master/.github/pull_request_template.md).  Notably, please include information about how the patch was tested.

File: hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/rpc/RpcClient.java [hunk 2]
```diff
@@ -638,6 +638,7 @@ public void createBucket(
           .setKeyName(bucketArgs.getEncryptionKey()).build();
     }
 
+
```
Comment (adoroszlai) [inline hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/rpc/RpcClient.java:641]: Nit: please avoid unnecessary whitespace changes.

File: hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/rpc/RpcClient.java (hunk 1 above)
Comment (chrisGe123) [reply hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/rpc/RpcClient.java:673]: Sure, I will make this change, thanks @adoroszlai

Comment (chrisGe123) [reply hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/rpc/RpcClient.java:673]: Can I just make the similar change like "layoutMsg" ? if the client doesn't set the replication, then we just mention "with server-side default bucket layout" ? @adoroszlai 

 String layoutMsg = bucketLayout != null
        ? "with bucket layout " + bucketLayout
        : "with server-side default bucket layout";

File: hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/rpc/RpcClient.java (hunk 2 above)
Comment (chrisGe123) [reply hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/rpc/RpcClient.java:641]: Sure, thanks @adoroszlai

File: hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/rpc/RpcClient.java (hunk 1 above)
Comment (adoroszlai) [reply hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/rpc/RpcClient.java:673]: @chrisGe123 sounds good, thanks for the suggestion

File: hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/rpc/RpcClient.java [hunk 3]
```diff
@@ -668,13 +668,21 @@ public void createBucket(
...
+        ? "with server-side default replication type"
+        : defaultReplicationConfig.getType().toString();
+
     String layoutMsg = bucketLayout != null
         ? "with bucket layout " + bucketLayout
         : "with server-side default bucket layout";
     LOG.info("Creating Bucket: {}/{}, {}, {} as owner, Versioning {}, " +
-            "Storage Type set to {} and Encryption set to {} ",
+            "Storage Type set to {} and Encryption set to {}, " +
+            "Replication Type set to {}, Namespace Quota set to {}, " +
```
Comment (adoroszlai) [inline hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/rpc/RpcClient.java:680]: Can you please tweak the replication-related part a bit?

1. `Replication Type set to with server-side default replication type` sounds awkward.
2. I think it should include not only replication type, but all details of the replication config, available via `configFormat()` of `ReplicationConfig`.  But I just realized `DefaultReplicationConfig` does not easily provide those details, so I created a PR (#4005) to expose them.  If that's merged, we can make the message more helpful.

**Comment by kerneltime** [comment]
@chrisGe123 is this just to improve the logging when creating a bucket via CLI?

File: hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/rpc/RpcClient.java (hunk 3 above)
Comment (chrisGe123) [reply hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/rpc/RpcClient.java:680]: Thanks for your comments @adoroszlai , I will modify this after you finish your changes

**Comment by chrisGe123** [comment]
> 

I believe so @kerneltime, all changes are just related to logging during bucket creation via CLI

File: hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/rpc/RpcClient.java (hunk 3 above)
Comment (adoroszlai) [reply hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/rpc/RpcClient.java:680]: Finally #4005 is merged, sorry for taking so long.

Comment (DaveTeng0) [reply hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/rpc/RpcClient.java:680]: Hey @chrisGe123 ! Just a heads-up that Attila has completed his changes~

Comment (chrisGe123) [reply hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/rpc/RpcClient.java:680]: thanks, @DaveTeng0 , I have finished the code changes but have some issues with my account of github, will submit this commit today or tomorrow.

Comment (chrisGe123) [reply hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/rpc/RpcClient.java:680]: @DaveTeng0 and @adoroszlai I am still waiting for Sammi to come back from holidays to help me out about github issue, but you guys can help me to review the test result as followed to see if this message change is ok:

```
# bin/ozone sh bucket create /vol1/bucket2 --layout FILE_SYSTEM_OPTIMIZED --type EC --replication rs-3-2-1024k
23/01/19 07:53:18 INFO rpc.RpcClient: Creating Bucket: vol1/bucket2, with bucket layout FILE_SYSTEM_OPTIMIZED, root as owner, Versioning false, Storage Type set to DISK and Encryption set to false, Replication Config is set to EC/ECReplicationConfig{data=3, parity=2, ecChunkSize=1048576, codec=rs}, Counts Quota set to -1, Space Quota set to -1 bytes


# bin/ozone sh bucket create /vol1/bucket3
23/01/19 07:54:33 INFO rpc.RpcClient: Creating Bucket: vol1/bucket3, with server-side default bucket layout, root as owner, Versioning false, Storage Type set to DISK and Encryption set to false, Replication Config is set to server-side default replication config, Counts Quota set to -1, Space Quota set to -1 bytes


# bin/ozone sh bucket create /vol1/bucket3 --layout OBJECT_STORE --replication-type EC --replication rs-6-3-1024k --namespace-quota=10 --space-quota=1073741824
23/01/19 07:55:27 INFO rpc.RpcClient: Creating Bucket: vol1/bucket3, with bucket layout OBJECT_STORE, root as owner, Versioning false, Storage Type set to DISK and Encryption set to false, Replication Config is set to EC/ECReplicationConfig{data=6, parity=3, ecChunkSize=1048576, codec=rs}, Counts Quota set to 10, Space Quota set to 1073741824 bytes
BUCKET_ALREADY_EXISTS Bucket already exist

# bin/ozone sh bucket create /vol1/bucket4 --layout OBJECT_STORE --replication-type EC --replication rs-6-3-1024k --namespace-quota=10 --space-quota=1073741824
23/01/19 07:55:48 INFO rpc.RpcClient: Creating Bucket: vol1/bucket4, with bucket layout OBJECT_STORE, root as owner, Versioning false, Storage Type set to DISK and Encryption set to false, Replication Config is set to EC/ECReplicationConfig{data=6, parity=3, ecChunkSize=1048576, codec=rs}, Counts Quota set to 10, Space Quota set to 1073741824 bytes
```

File: hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/rpc/RpcClient.java [hunk 4]
```diff
@@ -687,13 +687,21 @@ public void createBucket(
       builder.setDefaultReplicationConfig(defaultReplicationConfig);
     }
 
+    String replicationType = defaultReplicationConfig == null 
+        ? "with server-side default replication type"
```
Comment (adoroszlai) [inline hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/rpc/RpcClient.java:691]: I think this `with` is unnecessary, as the message would look like: `Replication Type set to with server-side default replication type`.

```suggestion
        ? "server-side default replication type"
```

**Review by adoroszlai — COMMENTED** [review]
Thanks @chrisGe123 for updating the patch.

File: hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/rpc/RpcClient.java (hunk 4 above)
Comment (chrisGe123) [reply hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/rpc/RpcClient.java:691]: This has been fixed in the latest commit

**Review by adoroszlai — APPROVED** [review]

File: hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/rpc/RpcClient.java (hunk 4 above)
Comment (adoroszlai) [reply hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/rpc/RpcClient.java]: Maybe it was fixed locally, but it didn't show up in the PR.  I have applied the suggestion since it's a trivial one.
